src/UI/Pages/FunctionPage/FunctionPage.tsx

```diff
@@ -19,43 +18,55 @@ const FunctionPage: React.FC<Props> = ({ viewportHeight }) => {
-        const {value} = target
+        const { value } = target
         setSearchValue(value)
     }
-    const handleKeySubmit: React.KeyboardEventHandler<HTMLInputElement> = function ({key}) {
+    const handleKeySubmit: React.KeyboardEventHandler<HTMLInputElement> = function ({ key }) {
         if (key == 'Enter') {
             handleSearchBook(searchValue)
         }
     }
+    useEffect(() => {
+        if (!isLoogedIn) {
+            navigate('/')
+        }
+    }, [isLoogedIn])
     return (
         <FunctionPageStyled viewportHeight={viewportHeight}>
             <HeaderSection />
             <div className="resultsContainer">
                 <div className="inputContainer">
                     <InputStyled>
-                        <input onKeyPress={handleKeySubmit} value={searchValue} onChange={handleChange} name="search" placeholder="Search by Author, Genre, year, title" />
+                        <input
+                            onKeyPress={handleKeySubmit}
+                            value={searchValue}
+                            onChange={handleChange}
+                            name="search"
+                            placeholder="Search by Author, Genre, year, title"
+                        />
                     </InputStyled>
                 </div>
-                <Title>All Results</Title>
+                <Title>All Results ({items.length})</Title>
 
                 <SubTitle>Sort by</SubTitle>
                 <div className="actionsContainer">
-                    <ButtonS onClick={() => setSelectedFilter('Filter')} showIcon={selectedFilter == 'Filter'}>
+                    <ButtonS onClick={setSelectedFilter} name={'Filter'} showIcon={selectedFilter == 'Filter'}>
                         Filter
                     </ButtonS>
-                    <ButtonS onClick={() => setSelectedFilter('Date')} showIcon={selectedFilter == 'Date'}>
+                    <ButtonS onClick={setSelectedFilter} name="Date" showIcon={selectedFilter == 'Date'}>
                         Date
                     </ButtonS>
-                    <ButtonS onClick={() => setSelectedFilter('Type')} showIcon={selectedFilter == 'Type'}>
+                    <ButtonS onClick={setSelectedFilter} name="Type" showIcon={selectedFilter == 'Type'}>
                         Type
                     </ButtonS>
-                    <ButtonS onClick={() => setSelectedFilter('Auther')} showIcon={selectedFilter == 'Auther'}>
+                    <ButtonS onClick={setSelectedFilter} name="Auther" showIcon={selectedFilter == 'Auther'}>
                         Auther
                     </ButtonS>
-                    <ButtonS onClick={() => setSelectedFilter('Category')} showIcon={selectedFilter == 'Category'}>
+                    <ButtonS onClick={setSelectedFilter} name="Category" showIcon={selectedFilter == 'Category'}>
                         Category
                     </ButtonS>
```

Comment:
Avoid code repetitiveness by declaring an array with the different props and map through this array here,
and for each item in this array, render a ButtonS component. 

Please ask Lior about this comment.